src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectState.cs

```diff
@@ -256,15 +308,9 @@ public ProjectState WithHostProject(HostProject hostProject)
             {
                 return this;
             }
-
-            var difference = ProjectDifference.ConfigurationChanged;
-            var documents = new Dictionary<string, DocumentState>(FilePathComparer.Instance);
-            foreach (var kvp in Documents)
-            {
-                documents.Add(kvp.Key, kvp.Value.WithConfigurationChange());
-            }
-
-            var state = new ProjectState(this, difference, hostProject, WorkspaceProject, documents);
+            
+            var documents = Documents.ToImmutableDictionary(kvp => kvp.Key, kvp => kvp.Value.WithConfigurationChange(), FilePathComparer.Instance);
+            var state = new ProjectState(this, ProjectDifference.ConfigurationChanged, hostProject, WorkspaceProject, documents, ImportsToRelatedDocuments);
```

Comment:
If the HostProject changed it means the configuration might have changed. That'd potentially impact the type of imports that exist in the project state since a new project engine is created. We need to re-calc `ImportsToRelatedDocuments` here as well.

Reply:
Yeah that's a good point 👍